shared/editor/nodes/ToggleBlock.ts

```diff
@@ -0,0 +1,605 @@
+                            deco.spec.fold === true
+                        )
+                      )
+                    )
+                  )
+                : undefined;
+
+            if (ancestor) {
+              const posAfterAncestorHead =
+                $cursor.start(depth(ancestor, $cursor)) +
+                ancestor.firstChild!.nodeSize;
+              const endOfAncestor = $cursor.end(depth(ancestor, $cursor));
+
+              if (
+                $cursor.pos > posAfterAncestorHead &&
+                $cursor.pos < endOfAncestor
+              ) {
+                tr.setMeta(ToggleBlock.pluginKey, {
+                  type: Action.UNFOLD,
+                  at: $cursor.before(depth(ancestor, $cursor)),
+                });
+              }
+            }
+          }
+        }
+
+        return tr;
+      },
+      props: {
+        decorations: (state) => ToggleBlock.pluginKey.getState(state),
+        nodeViews: {
+          [this.name]: (node, view, getPos, decorations, innerDecorations) =>
+            new ToggleBlockView(
+              node,
+              view,
+              getPos,
+              decorations,
+              innerDecorations,
+              this.editor.props
+            ),
+        },
+      },
+    });
+
+    return [initPlugin, foldPlugin];
+  }
+
+  get rulePlugins() {
+    return [toggleBlocksRule];
+  }
+
+  keys({ type }: { type: NodeType }): Record<string, Command> {
+    return {
+      Backspace: chainCommands(lift, (state, dispatch) => {
+        const { $cursor } = state.selection as TextSelection;
+        if (!$cursor) {
+          return false;
+        }
+
+        if (!$cursor.node().isTextblock) {
+          return false;
+        }
+
+        const $cut = state.doc.resolve($cursor.before());
+
+        if (!$cut.nodeBefore || $cut.nodeBefore.type.name !== this.name) {
+          return false;
+        }
+
+        return joinTextblockBackward(state, dispatch);
+      }),
+      Enter: chainCommands(createParagraphBefore, split, (state, dispatch) => {
+        const { $from } = state.selection;
+        const parent = $from.node($from.depth - 1);
+        if (parent.type.name !== this.name) {
+          return false;
+        }
+
+        // if cursor lies within immediate first child, ignore the handling here
+        if ($from.index($from.depth - 1) === 0) {
+          return false;
+        }
+
+        return chainTransactions(
+          newlineInCode,
+          createParagraphNear,
+          splitBlock
+        )(state, dispatch);
+      }),
+      Delete: (state, dispatch) =>
+        chainTransactions(liftNext, joinForward)(state, dispatch),
+      Tab: sinkBlockInto(type),
+      "Shift-Tab": liftLastBlockOutOf(type),
+    };
+  }
+
+  commands({ type }: { type: NodeType }) {
+    return () => wrapIn(type);
+  }
+
+  toMarkdown(state: MarkdownSerializerState, node: ProsemirrorNode) {
+    state.write(state.repeat("+", 3 + height(node)) + "\n");
```

Comment:
Is this right? I'd expect the fence to always be `+++`

Reply:
_Think [nesting](https://github.com/markdown-it/markdown-it-container/issues/33#issuecomment-551111227)..._